forwarder/kernel-forwarder/pkg/kernelforwarder/kernelforwarder.go

```diff
@@ -59,7 +66,13 @@ func (k *KernelForwarder) CreateForwarderServer(config *common.ForwarderConfig)
 // Request handler for connections
 func (k *KernelForwarder) Request(ctx context.Context, crossConnect *crossconnect.CrossConnect) (*crossconnect.CrossConnect, error) {
 	logrus.Infof("Request() called with %v", crossConnect)
-	err := k.connectOrDisconnect(crossConnect, cCONNECT)
+
+	if err := crossConnect.IsValid(); err != nil {
+		logrus.Errorf("Close: %v is not valid, reason: %v", crossConnect, err)
+		return crossConnect, err
+	}
+
+	err := k.connectOrDisconnect(crossConnect, true)
```

Comment:
@artem-belov 
Is it possible to make it more explicit here? 
I'm open to use something else and ditch the connect/disconnect labels as long as it's better than true/false 👍

Reply:
I also think we can solve this problem on kernel-sdk level.